File: app/src/main/java/com/dcurreli/spese/adapters/SaldoAdapter.java
package com.dcurreli.spese.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.dcurreli.spese.R;
import com.dcurreli.spese.objects.Spesa;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class SaldoAdapter extends RecyclerView.Adapter<SaldoAdapter.MyViewHolder> {

    private Context context;
    private final ArrayList<Spesa> speseList;

    public SaldoAdapter(Context context, ArrayList<Spesa> speseList) {
        this.context = context;
        this.speseList = speseList;
    }

    public Context getContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.load_spese_tab_saldo_item, parent, false);
        return new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        Spesa spesa = speseList.get(position);

        holder.pagatore.setText(spesa.getPagatore());
        holder.importo.setText(spesa.importoAsTextEuro());

    }

    @Override
    public int getItemCount() {
        return speseList.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView pagatore, importo;

        public MyViewHolder(@NotNull View itemView) {
            super(itemView);

            pagatore = itemView.findViewById(R.id.saldo_pagatore);
            importo = itemView.findViewById(R.id.saldo_importo);

        }
    }
}

File: app/src/main/java/com/dcurreli/spese/adapters/SaldoAdapter.kt
package com.dcurreli.spese.adapters

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.dcurreli.spese.R
import com.dcurreli.spese.data.entity.Spesa

class SaldoAdapter : RecyclerView.Adapter<SaldoAdapter.ViewHolder>() {

    private val speseList = ArrayList<Spesa>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.load_spese_tab_saldo_totali_item, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val spesa = speseList[position]
        holder.pagatore.text = spesa.pagatore
        holder.importo.text = spesa.importoAsTextEuro()

    }

    // return the number of the items in the list
    override fun getItemCount(): Int {
        return speseList.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun addItems(newSpesaList : List<Spesa>){
        speseList.clear()
        speseList.addAll(newSpesaList)
        notifyDataSetChanged()
    }

    // Holds the views for adding it to image and text
    class ViewHolder(ItemView: View) : RecyclerView.ViewHolder(ItemView) {
        val pagatore: TextView = itemView.findViewById(R.id.saldo_pagatore)
        val importo: TextView = itemView.findViewById(R.id.saldo_importo)
    }
}